weasyprint/formatting_structure/boxes.py

```diff
@@ -336,6 +336,12 @@ def is_input(self):
                 return not isinstance(self, (LineBox, TextBox))
         return False
 
+    def is_form(self):
+        """Return whether this box is a form element."""
+        if self.element is None:
+            return False
+        return self.element.tag in 'form'
+
```

Comment:
Does a `form` not need to consider the "appearance == auto"? 

[I was treating](https://github.com/alenards/WeasyPrint/pull/1/files#diff-24ffb67744f69f2cd431fe25d989592e2629e7ef6da04e6e15188da933272ab5R313-R315) the form like it was similar to the `input` predicate open at: 

https://github.com/Kozea/WeasyPrint/pull/2143/files#diff-24ffb67744f69f2cd431fe25d989592e2629e7ef6da04e6e15188da933272ab5L334-L337

Maybe I didn't catch that point in my reading of the PDF spec.

Reply:
The `appearance` property is actually from CSS, not PDF. `boxes.py` in general isn't aware of PDFs, yet - it's still at the extracting elements from HTML stage.

That aside, I don't think we need to consider appearance for forms, though - [appearance](https://developer.mozilla.org/en-US/docs/Web/CSS/appearance#compat-auto) seems to affect input-level elements underneath the form. If a form only has inputs with `appearance != auto`, it'll be represented as an empty list in the `forms` dictionary, meaning nothing will be rendered.

Kind of a confusing feature because of its history. There's a note about its use on modern websites:

<img width="638" alt="image" src="https://github.com/Kozea/WeasyPrint/assets/15126660/0aabc1a8-7df6-45fe-8997-7fd88260075d">